backend/sql/init.sql

```diff
@@ -0,0 +1,172 @@
+-- Ongoing project in a company.
+-- Experiment data is grouped into projects, i. e. each experiment fully
+-- belongs to a single project.
+-- Moreover, all experiments from one experiment data file belong to one project.
+create table if not exists project
+(
+    project_id    serial    not null primary key,
+    name          text      not null unique,
+    description   text      null,
+    creation_date timestamp not null,
+    last_update   timestamp not null
+);
+
+-- Describes how signals were measured and what they denote (their semantics).
+-- All experiments from one experiment data file have the same methodology.
+create table if not exists test_methodology
+(
+    test_methodology_id serial        not null primary key,
+    name                text          not null unique,
+    description         text          null,
+    confluence_link     varchar(1000) null
+);
+
+-- Something that reacts with compounds that scientists want to research.
+-- The goal of their work is to check how a compound affects a target.
+create table if not exists target
+(
+    target_id       serial      not null primary key,
+    name            text        not null unique,
+    creation_date   timestamp   not null
+);
+
+create table if not exists compound
+(
+    compound_id     serial          not null primary key,
+    name            text            not null unique,
+    creation_date   timestamp       not null,
+    chemsoft_link   varchar(1000)   null
+);
+
+-- Corresponds to an Excel file.
+-- Here we store data that is shared by all experiments in one file,
+-- namely methodology and project.
+-- Additionally we store file name and binary contents because
+-- it should be possible to download these files.
+create table if not exists experiment_file
+(
+    experiment_file_id serial not null primary key,
+    project_id         int       not null,
+    methodology_id     int       null,  -- normally not null, but methodology can be removed
+    upload_date        timestamp not null,
+    -- Arbitrary metadata from the file
+    meta               json      null,
+
+    -- Name of the file
+    name               text      not null,
+    -- .xlsx blob
+    contents           bytea     not null,
```

Comment:
Hm, I was thinking a bit about it, but not too much. Since you asked, I tried to google a bit. It looks like a controversial topic without a single right answer. AFAIU it can be split into 2 questions:
1. Should we store files in DB or FS? We can store some _pointers_ in DB and store files as normal files of course.
2. PostgreSQL-specific question: if we decide to store in DB, should we use `bytea` or something else?

I think the primary criterion for the first question is the size of files we are going to store. The largest file we have now has 30270 bytes. I don't know how large files we should expect in practice, that's an open question, but probably we can set some reasonable limits ourselves saying "it's just MVP". So I think it's fine to say everything works fine on files less than 240kb, that's enough to store 8 times more data than the largest example we have. And probably average file will be ≈10-30kb.

Now the question is whether 10-30kb on average and 240kb in extreme case is a lot or not. In a [random old paper from 2006](https://www.microsoft.com/en-us/research/publication/to-blob-or-not-to-blob-large-object-storage-in-a-database-or-a-filesystem/?from=http%3A%2F%2Fresearch.microsoft.com%2Fapps%2Fpubs%2Fdefault.aspx%3Fid%3D64525) they say:
> As expected from the common wisdom, objects smaller than 256K are best stored in a database while objects larger than 1M are best stored in the filesystem. 

Of course it's old data and not about PostgreSQL and each case is unique and so on. But I think 256K is a reasonable size to distinguish "data that is not too big to be stored as blob in DB" and "data that is too big".

Ideally we should do some performance testing and make a decision based on that. At this stage I would just pick the simplest option and I _guess_ that storing files as blobs in DB is simpler, you can just access them directly from DB instead of accessing pointers and then reading from FS. And there is no need to think about consistency between files and pointers. There is one SO [comment](https://softwareengineering.stackexchange.com/a/150816) that I liked.

> I would be pragmatic about it, and follow the "don't optimize yet" principle.

> storing the data in the database does relieve you from dealing with other problems, e.g. where should the files be stored, how should they be backed up, etc.

> I would personally choose to store the data in the database, but make sure that the BLOBS are not read until they are really needed, i.e. no "SELECT * FROM ..." executed on those tables containing blogs.

All of that makes sense to me in our case. So, unless you have a good reason why you think that storing files differently will be _easier_, I would store them this way.

Regarding `bytea` specifically, I googled a bit and apparently there are two options provided by PostgreSQL:
1. `bytea`: https://www.postgresql.org/docs/9.0/datatype-binary.html. They say: "The SQL standard defines a different binary string type, called BLOB or BINARY LARGE OBJECT. The input format is different from bytea, but the provided functions and operators are mostly the same". Sounds appropriate in our case.
2. Streaming approach: https://www.postgresql.org/docs/current/largeobjects.html

I wouldn't bother with streaming prematurely, it's more like an optimization.

Reply:
Nice, thanks